app/services/DateService.scala

```diff
@@ -19,42 +19,49 @@ package services
 import com.google.inject.ImplementedBy
 import config.FrontendAppConfig
 
-import java.time.LocalDate
+import java.time.{Clock, Duration, LocalDate, ZoneOffset}
 import java.time.Month.APRIL
 import javax.inject.{Inject, Singleton}
 
 @Singleton
 class DateService @Inject()(implicit val frontendAppConfig: FrontendAppConfig) extends DateServiceInterface {
 
-  def getCurrentDate(isTimeMachineEnabled: Boolean = false): LocalDate = {
-    if (isTimeMachineEnabled) {
-      frontendAppConfig
-        .timeMachineAddYears.map(LocalDate.now().plusYears(_))
-        .getOrElse(LocalDate.now())
+  // TODO: instantiate clock depending on the configuration settings
+  private val timeMachineIsOn: Boolean = frontendAppConfig.timeMachineAddYears.nonEmpty
+
+  lazy val clock: Clock = if (timeMachineIsOn) {
```

Comment:
What is the benefit of using the clock here rather than keeping Localdate.now() as it is (and if time machine is on then add a year)?

Reply:
this is yet another way to simulate fixed time or implement timeMachine, have a look at the method which Clock instance can provide, you will see what can be done, ie. for unit tests etc

Reply:
Yep, I think it's a good idea. Do we have a separate ticket for implementing this?